resolwe/flow/executors/__main__.py

```diff
@@ -98,9 +117,16 @@ async def run_executor():
 
     async def _sequential():
         """Run some things sequentially but asynchronously."""
-        await transfer_data()
-        await run_executor()
-        await collect_files()
+        try:
+            # Try to obtains exclusive lock over stdout and jsonout files.
+            # When lock can not be obtained this task is probably a duplicate
+            # spawned by celery so lock error and skip processing.
+            log_file, json_file = get_stdout_json_file()
+            await transfer_data()
+            await run_executor(log_file, json_file)
+            await collect_files()
+        except FileExistsError:
+            logger.error("Stdout or jsonout file already exists, aborting.")
```

Comment:
Do you need to send `ABORT` command to the listener to keep the semaphore in sync?

Reply:
I believe not since executors has not even started at this point and no messages have been exchanged.

Reply:
I will take another look into that.

Reply:
It is increased here: https://github.com/genialis/resolwe/blob/fcbf83ac008d5fc039f4e0e69c00fc2e4b439fc0/resolwe/flow/managers/dispatcher.py#L429

Reply:
Yes, it is. So to keep the counters in sync the abort has to be called.